sdk/typescript/src/signers/txn-data-serializers/call-arg-serializer.ts

```diff
@@ -54,8 +54,9 @@ export class CallArgSerializer {
 
   async newObjectArg(objectId: string): Promise<ObjectArg> {
     const object = await this.provider.getObject(objectId);
-    if (isSharedObject(object)) {
-      return { Shared: objectId };
+    const initial_shared_version = isSharedObject(object);
+    if (initial_shared_version) {
+      return { Shared: { objectId, initial_shared_version } };
     }
```

Comment:
This change will break a lot of wallet & DApps that rely on shared objects. After https://github.com/MystenLabs/sui/pull/5350 is merged, can you do

```suggestion
   const isNewSharedObjectAPI = this.provider.getRpcApiVersion() === '0.12.0';
   if (!isNewSharedObjectAPI && isSharedObject(object)) {
           return { Shared: objectId };
   } else if (isNewSharedObjectAPI) {
   	const initial_shared_version = isSharedObject(object);
        if (initial_shared_version) {
            return { Shared: { objectId, initial_shared_version } };
        }
   }
    
```
?

Reply:
@666lcz, if I understand correctly, this change will allow a wallet/DApp that has upgraded to talk to a pre-0.12.0 full node, but an older DApp talking to a newer full node would currently still fail, is that right?

If that's the case, how likely is that scenario at the moment (i.e. for how long will we have newer DApps talking to older nodes)?  I was under the impression that we upgraded the full nodes first, and then the SDK is updated, or is that not true?